docs/standard/serialization/system-text-json-supported-collection-types.md

```diff
@@ -0,0 +1,310 @@
+
+### Custom collections with deserialization support
+
+A custom collection is supported for deserialization if it:
+
+::: zone pivot="dotnet-5-0"
+
+* Isn't an interface or abstract.
+* Has a parameterless constructor.
+* Contains element types that are supported by <xref:System.Text.Json.JsonSerializer>.
+* Implements or inherits one or more of the following interfaces or classes:
+  * <xref:System.Collections.Concurrent.ConcurrentQueue%601>
+  * <xref:System.Collections.Concurrent.ConcurrentStack%601> \*
+  * <xref:System.Collections.Generic.ICollection%601>
+  * <xref:System.Collections.IDictionary>
+  * <xref:System.Collections.Generic.IDictionary%602>
+  * <xref:System.Collections.IList>
+  * <xref:System.Collections.Generic.IList%601>
+  * <xref:System.Collections.Queue>
+  * <xref:System.Collections.Generic.Queue%601>
+  * <xref:System.Collections.Stack> \*
+  * <xref:System.Collections.Generic.Stack%601> \*
+
+::: zone-end
+
+::: zone pivot="dotnet-core-3-1"
+
+* Isn't an interface or abstract.
+* Has a parameterless constructor.
+* Contains element types that are supported by <xref:System.Text.Json.JsonSerializer>.
+* Implements or inherits one or more of the following interfaces or classes:
+  * <xref:System.Collections.Concurrent.ConcurrentQueue%601>
+  * <xref:System.Collections.Concurrent.ConcurrentStack%601> \*
+  * <xref:System.Collections.Generic.ICollection%601>
+  * <xref:System.Collections.IDictionary>
+  * [IDictionary\<string, TValue>](xref:System.Collections.Generic.IDictionary%602) \*\*
+  * <xref:System.Collections.IList>
+  * <xref:System.Collections.Generic.IList%601>
+  * <xref:System.Collections.Queue>
+  * <xref:System.Collections.Generic.Queue%601>
+  * <xref:System.Collections.Stack> \*
+  * <xref:System.Collections.Generic.Stack%601> \*
+
+\*\* Non-`string` keys are not supported in .NET Core 3.1.
+
+::: zone-end
+
+\* See [Support round trip for Stack\<T>](system-text-json-converters-how-to.md#support-round-trip-for-stackt).
+
+### Custom collections with known issues
+
+There are known issues with the following custom collections:
+
+- <xref:System.Dynamic.ExpandoObject>: See [dotnet/runtime#29690](https://github.com/dotnet/runtime/issues/29690).
+- <xref:System.Dynamic.DynamicObject>: See [dotnet/runtime#1808](https://github.com/dotnet/runtime/issues/1808).
+- <xref:System.Data.DataTable>: See [dotnet/docs#21366](https://github.com/dotnet/docs/issues/21366).
+- <xref:Microsoft.AspNetCore.Http.FormFile?displayProperty=fullName>: See [dotnet/runtime#1559](https://github.com/dotnet/runtime/issues/1559).
+- <xref:Microsoft.AspNetCore.Http.IFormCollection?displayProperty=fullName>: See [dotnet/runtime#1559](https://github.com/dotnet/runtime/issues/1559).
+
+## See also
+
+For more information about known issues, see the [open issues in System.Text.Json](https://github.com/dotnet/runtime/issues?q=is%3Aopen+is%3Aissue+label%3Aarea-System.Text.Json).
```

Comment:
Shouldn't this sentence be moved to the end of the previous section about known issues?